src/docs/guide/usage/formatter.md

```diff
@@ -67,49 +67,65 @@ $ bun add -D oxfmt
+By default, `oxfmt` automatically tries to find the nearest `.oxfmtrc.json` or `.oxfmtrc.jsonc` file from the current working directory. If not found, the default configuration options are used.
 
-Also you can specify your config file by `-c yourconfig.jsonc` flag.
+You can also specify your config file with the `-c yourconfig.jsonc` flag.
 
-Almost all format options are compatible with Prettier's [options](https://prettier.io/docs/options).
-So you may finish your setup by just renaming `.prettierrc.json` to `.oxfmtrc.jsonc`.
+Almost all formatting options are compatible with Prettier's [options](https://prettier.io/docs/options).
+So you can migrate from Prettier by simply renaming `.prettierrc.json` to `.oxfmtrc.jsonc`.
 
-Config file follows this schema.
-
-> https://github.com/oxc-project/oxc/blob/main/npm/oxfmt/configuration_schema.json
-
-You can start with this template if your editor is supporting JSON Schema.
+We also recommend adding the `$schema` to the config file after copying it:
 
 ```json
 {
   "$schema": "./node_modules/oxfmt/configuration_schema.json"
 }
 ```
 
+You can see the full JSON schema for the config file here:
+
+> https://github.com/oxc-project/oxc/blob/main/npm/oxfmt/configuration_schema.json
+
 ## Ignore file
 
-By default, `oxfmt` automatically finds the `.gitignore` and `.prettierignore` file from current working directory.
+By default, `oxfmt` automatically finds the `.gitignore` and `.prettierignore` files from the current working directory.
 
 Also you can specify your ignore file by `--ignore-path your.ignore` flag.
 
 VCS directories like `.git` and `.svn` are always ignored. Also global and nested ignores are not respected.
 
-In addition, `.oxfmtrc.json(c)` also supports `ignorePatterns` field.
+In addition, `.oxfmtrc.json(c)` supports an `ignorePatterns` field.
+
+## Pre-commit hook
+
+If you want to auto-format staged files with oxfmt in a git pre-commit hook, you can use `oxfmt --no-error-on-unmatched-pattern`.
+
+This command is equivalent to `"prettier --ignore-unknown --write"` and will format all files that are supported by oxfmt. The `--no-error-on-unmatched-pattern` flag ensures that oxfmt will not raise errors if there are no supported files passed in by your pre-commit hook tool (e.g. only Ruby files are staged).
+
+You can also pass `--check` to only check the formatting of files, without modifying them in the pre-commit hook.
+
+If you are using a pre-commit hook via husky/lint-staged, you can run oxfmt with it by updating your package.json like so:
+
+```json
+"lint-staged": {
+  "*": "oxfmt --no-error-on-unmatched-pattern"
+},
+```
 
 ## FAQs
 
 ### What kinds of files are supported?
 
-Currently, only JS and TS files are supported.
-JSX is always available in JS files, but for TS files, the `.tsx` extension is required to use JSX.
+Currently, only JavaScript and TypeScript files are supported (`.js`, `.jsx`, `.ts`, `.tsx`, `.mjs`, `.cjs`, `.mts`, and `.cts`).
```

Comment:
It might be a bit tricky to explicitly list all file extensions here.

That's because we recognize various other files as JS/TS beyond what's listed here.

https://github.com/oxc-project/oxc/blob/5da1a6326d0a2d8ed0d98901acf098131aad5f4e/crates/oxc_formatter/src/service/parse_utils.rs#L17

Reply:
Maybe this phrasing works better?:

```suggestion
Currently, only JavaScript and TypeScript files are supported (using all common JS/TS extensions and filenames, including `.js`, `.jsx`, `.ts`, `.mjs`, etc.).
```

Main goal here is just to communicate that we did not forget `mjs`/`cjs`-style file names :)